### ISSUE TITLE
Cant connect to vscode server although could connect over SSH on Putty.

### ISSUE THREAD

**Issue by bhosalems** [microsoft/vscode-remote-release#8736]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.80.0 (user setup)
- OS Version: Windows_NT x64 10.0.22621

Steps to Reproduce:
Try to connect with the server 

Some latest log 
```
> rm: cannot remove '/home/csgrad/mbhosale/.vscode-server/.704ed70d4fd1c6bd6342c43
> 36f1ede30d1cff4710.log': No such file or directory
[23:33:09.276] > 
> Found old VS Code install 660393deaaa6d1996740ff4880f1bad43768c814, attempting t
> to clean up
> Failed to acquire lock for install, not deleting: 660393deaaa6d1996740ff4880f1ba
> ad43768c814
> Found old VS Code install b3e4e68a0bc097f0ae7907b217c1119af9e03435, attempting t
> to clean up
[23:33:09.291] > 
[23:33:09.346] > Install still has running processes, not deleting: b3e4e68a0bc097f0ae7907b217c11
> 119af9e03435
> Found old VS Code install 8fa188b2b301d36553cbc9ce1b0a146ccb93351f, attempting t
> to clean up
[23:33:09.362] > Deleting old install from /home/csgrad/mbhosale/.vscode-server/bin/8fa188b2b301d
> d36553cbc9ce1b0a146ccb93351f
> rm: cannot remove '/home/csgrad/mbhosale/.vscode-server/bin/8fa188b2b301d36553cb
> bc9ce1b0a146ccb93351f': Directory not empty
> Deleting old log file from /home/csgrad/mbhosale/.vscode-server/.8fa188b2b301d36
> 6553cbc9ce1b0a146ccb93351f.log
> rm: cannot remove '/home/csgrad/mbhosale/.vscode-server/.8fa188b2b301d36553cbc9c
> ce1b0a146ccb93351f.log': No such file or directory
> Found old VS Code install 1ad8d514439d5077d2b0b7ee64d2ce82a9308e5a, attempting t
> to clean up
[23:33:09.380] > 
[23:33:09.416] > Deleting old install from /home/csgrad/mbhosale/.vscode-server/bin/1ad8d514439d5
> 5077d2b0b7ee64d2ce82a9308e5a
> rm: cannot remove '/home/csgrad/mbhosale/.vscode-server/bin/1ad8d514439d5077d2b0
> 0b7ee64d2ce82a9308e5a'Deleting old log file from /home/csgrad/mbhosale/.vscode-se
> erver/.1ad8d514439d5077d2b0b7ee64d2ce82a9308e5a.log
> : Directory not empty
> rm: cannot remove '/home/csgrad/mbhosale/.vscode-server/.1ad8d514439d5077d2b0b7e
> ee64d2ce82a9308e5a.log': No such file or directory
> Found old VS Code install 695af097c7bd098fbf017ce3ac85e09bbc5dda06, attempting t
> to clean up
[23:33:09.426] > 
[23:33:09.472] > Deleting old install from /home/csgrad/mbhosale/.vscode-server/bin/695af097c7bd0
> 098fbf017ce3ac85e09bbc5dda06
> rm: cannot remove '/home/csgrad/mbhosale/.vscode-server/bin/695af097c7bd098fbf01
> 17ce3ac85e09bbc5dda06'Deleting old log file from /home/csgrad/mbhosale/.vscode-se
> erver/.695af097c7bd098fbf017ce3ac85e09bbc5dda06.log
> : Directory not empty
> rm: cannot remove '/home/csgrad/mbhosale/.vscode-server/.695af097c7bd098fbf017ce
> e3ac85e09bbc5dda06.log': No such file or directory
> Found existing installation at /home/csgrad/mbhosale/.vscode-server/bin/660393de
> eaaa6d1996740ff4880f1bad43768c814...
[23:33:09.478] > 
> Checking /home/csgrad/mbhosale/.vscode-server/.660393deaaa6d1996740ff4880f1bad43
> 3768c814.log and /home/csgrad/mbhosale/.vscode-server/.660393deaaa6d1996740ff4880
> 0f1bad43768c814.pid for a running server
> Looking for server with pid: 3477936
[23:33:09.494] > 
[23:33:09.521] > Found running server...
> 
> *
> * Visual Studio Code Server
> *
> * By using the software, you agree to
> * the Visual Studio Code Server License Terms (https://aka.ms/vscode-server-lice
> ense) and
> * the Microsoft Privacy Statement (https://privacy.microsoft.com/en-US/privacyst
> tatement).
> *
> 
[23:33:09.534] > 
> Checking server status on port 45231 with wget
[23:33:09.545] > b668c153fe8c: start
> SSH_AUTH_SOCK====
> DISPLAY====
> webUiAccessToken====
> listeningOn==45231==
> osReleaseId==ubuntu==
> arch==x86_64==
> vscodeArch==x64==
> bitness==64==
> tmpDir==/run/user/688020==
> platform==linux==
> unpackResult====
> didLocalDownload==0==
> downloadTime====
> installTime====
> extInstallTime====
> serverStartTime====
> connectionToken==1aaaaa11-11a1-111a-1a11-a111a11a1111==
> b668c153fe8c: end
[23:33:09.560] > 
```

**Comment by bhosalems** [comment]
Can someone please take a look into it ?

**Comment by qideng7** [comment]
Similar issue. Machine A could ssh to machine B , and remotely access Jupyter notebook/lab. But vscode (newest version, 1.80.1) tunnel was not working (log as below). But the other way around worked, i.e., vscode on machine B can tunnel access vscode server on machine A. 


[19:41:10.408] Log Level: 2
[19:41:10.419] SSH Resolver called for "ssh-remote+129.81.7.23", attempt 1
[19:41:10.419] "remote.SSH.useLocalServer": false
[19:41:10.419] "remote.SSH.showLoginTerminal": false
[19:41:10.419] "remote.SSH.remotePlatform": {}
[19:41:10.419] "remote.SSH.path": undefined
[19:41:10.420] "remote.SSH.configFile": undefined
[19:41:10.420] "remote.SSH.useFlock": true
[19:41:10.420] "remote.SSH.lockfilesInTmp": false
[19:41:10.420] "remote.SSH.localServerDownload": auto
[19:41:10.420] "remote.SSH.remoteServerListenOnSocket": false
[19:41:10.420] "remote.SSH.showLoginTerminal": false
[19:41:10.420] "remote.SSH.defaultExtensions": []
[19:41:10.420] "remote.SSH.loglevel": 2
[19:41:10.421] "remote.SSH.enableDynamicForwarding": true
[19:41:10.421] "remote.SSH.enableRemoteCommand": false
[19:41:10.422] "remote.SSH.serverPickPortsFromRange": {}
[19:41:10.422] "remote.SSH.serverInstallPath": {}
[19:41:10.426] VS Code version: 1.80.1
[19:41:10.426] Remote-SSH version: remote-ssh@0.102.0
[19:41:10.426] win32 x64
[19:41:10.427] SSH Resolver called for host: 129.81.7.23
[19:41:10.427] Setting up SSH remote "129.81.7.23"
[19:41:10.432] Using commit id "74f6148eb9ea00507ec113ec51c489d6ffb4b771" and quality "stable" for server
[19:41:10.436] Install and start server if needed
[19:41:11.852] Checking ssh with "C:\Python311\Scripts\ssh.exe -V"
[19:41:11.855] Got error from ssh: spawn C:\Python311\Scripts\ssh.exe ENOENT
[19:41:11.855] Checking ssh with "C:\Python311\ssh.exe -V"
[19:41:11.857] Got error from ssh: spawn C:\Python311\ssh.exe ENOENT
[19:41:11.857] Checking ssh with "C:\Windows\system32\ssh.exe -V"
[19:41:11.858] Got error from ssh: spawn C:\Windows\system32\ssh.exe ENOENT
[19:41:11.859] Checking ssh with "C:\Windows\ssh.exe -V"
[19:41:11.860] Got error from ssh: spawn C:\Windows\ssh.exe ENOENT
[19:41:11.860] Checking ssh with "C:\Windows\System32\Wbem\ssh.exe -V"
[19:41:11.861] Got error from ssh: spawn C:\Windows\System32\Wbem\ssh.exe ENOENT
[19:41:11.861] Checking ssh with "C:\Windows\System32\WindowsPowerShell\v1.0\ssh.exe -V"
[19:41:11.863] Got error from ssh: spawn C:\Windows\System32\WindowsPowerShell\v1.0\ssh.exe ENOENT
[19:41:11.863] Checking ssh with "C:\Windows\System32\OpenSSH\ssh.exe -V"
[19:41:11.916] > OpenSSH_for_Windows_8.1p1, LibreSSL 3.0.2

[19:41:11.921] Remote command length: 6211/8192 characters
[19:41:11.921] Running script with connection command: "C:\Windows\System32\OpenSSH\ssh.exe" -T -D 11306 "129.81.7.23" powershell
[19:41:11.924] Terminal shell path: C:\Windows\System32\cmd.exe
[19:41:12.163] > ]0;C:\Windows\System32\cmd.exe
[19:41:12.163] Got some output, clearing connection timeout
[19:41:13.062] > tu-ske7xh5\rs1@129.81.7.23's password: 
[19:41:13.062] Showing password prompt
[19:41:23.198] Got password response
[19:41:23.198] "install" wrote data to terminal: "***********"
[19:41:23.212] > 
[19:41:24.192] > . : File C:\Users\rs1\Documents\WindowsPowerShell\profile.ps1 cannot be loade
> d because running
[19:41:24.209] > scripts is disabled on this system. For more information, see about_Execution_Po
> licies at
> https:/go.microsoft.com/fwlink/?LinkID=135170.
> At line:1 char:3
> + . 'C:\Users\rs1\Documents\WindowsPowerShell\profile.ps1'
> +   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : SecurityError: (:) [], PSSecurityException        
>     + FullyQualifiedErrorId : UnauthorizedAccess
[19:41:24.379] > Windows PowerShell
> Copyright (C) Microsoft Corporation. All rights reserved.
> 
> Install the latest PowerShell for new features and improvements! https://aka.ms/
> PSWindows
[19:41:24.395] > 
[19:41:24.679] > . : File C:\Users\rs1\Documents\WindowsPowerShell\profile.ps1 cannot be loade
> d because running
[19:41:24.690] > scripts is disabled on this system. For more information, see about_Execution_Po
> licies at
> https:/go.microsoft.com/fwlink/?LinkID=135170.
> At line:1 char:3
> + . 'C:\Users\rs1\Documents\WindowsPowerShell\profile.ps1'
> +   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : SecurityError: (:) [], PSSecurityException        
>     + FullyQualifiedErrorId : UnauthorizedAccess
[19:41:24.707] > PS C:\Users\rs1> 
[19:41:24.767] > 
[19:41:24.786] > PS C:\Users\rs1> $uuid="ebebd4e708bc"
> PS C:\Users\rs1> "${uuid}: running"  
> ebebd4e708bc: running
[19:41:24.794] > 
> PS C:\Users\rs1> "ebebd4e708bc: pauseLog"
> ebebd4e708bc: pauseLog
[19:41:25.523] > 
> PS C:\Users\rs1> main
[19:41:25.533] > 
[19:41:26.747] > Acquiring lock on C:\Users\rs1\.vscode-server\bin\74f6148eb9ea00507ec113ec51c
> 489d6ffb4b771\vscode-remote-lock.74f6148eb9ea00507ec113ec51c489d6ffb4b771       
[19:41:26.753] > 
> Looking for existing server in C:\Users\rs1\.vscode-server\bin\74f6148eb9ea00
> 507ec113ec51c489d6ffb4b771
> Found installed server
[19:41:26.764] > 
[19:41:26.790] > 
> 
>     Directory: C:\Users\rs1\.vscode-server
> 
> 
> Mode                 LastWriteTime         Length Name
> 
> ----                 -------------         ------ ----
[19:41:26.797] > 
> 
> -a----         7/22/2023   7:41 PM              0 .74f6148eb9ea00507ec113ec51c48
> 9d6ffb4b771.token
[19:41:26.813] > 
[19:41:26.849] > Successfully processed 1 files; Failed processing 0 files
[19:41:26.901] > Successfully processed 1 files; Failed processing 0 files
[19:41:26.946] > Successfully processed 1 files; Failed processing 0 files
[19:41:26.988] > Successfully processed 1 files; Failed processing 0 files
[19:41:26.999] > Starting server: & 'C:\Users\rs1\.vscode-server\bin\74f6148eb9ea00507ec113ec5
> 1c489d6ffb4b771\bin\code-server.cmd' --start-server --host=127.0.0.1 --accept-se
> rver-license-terms --enable-remote-auto-shutdown --port=0 --connection-token-fil
> e 'C:\Users\rs1\.vscode-server\.74f6148eb9ea00507ec113ec51c489d6ffb4b771.toke
> n' --telemetry-level all  *> 'C:\Users\rs1\.vscode-server\.74f6148eb9ea00507e
> c113ec51c489d6ffb4b771.log'
[19:41:27.014] > 
[19:41:31.195] > Server did not start successfully. Full server log at C:\Users\rs1\.vscode-se
> rver\.74f6148eb9ea00507ec113ec51c489d6ffb4b771.log >>>
[19:41:31.201] > 
> <<< End of server log
[19:41:31.208] > 
> ebebd4e708bc: start
> exitCode==32==
> listeningOn====
> osReleaseId==windows==
> osVersion====
> arch==x64==
[19:41:31.217] > 
> platform==windows==
> unpackResult====
> didLocalDownload==False==
> downloadTime====
> installTime====
> extInstallTime====
> serverStartTime==405==
[19:41:31.224] > 
> ebebd4e708bc: end
[19:41:31.225] Received install output: 
exitCode==32==
listeningOn====
osReleaseId==windows==
osVersion====
arch==x64==
platform==windows==
unpackResult====
didLocalDownload==False==
downloadTime====
installTime====
extInstallTime====
serverStartTime==405==

[19:41:31.229] Resolver error: Error: The VS Code Server failed to start
	at m.ServerInstallError (c:\Users\rs1\.vscode\extensions\ms-vscode-remote.remote-ssh-0.102.0\out\extension.js:1:584079)
	at f (c:\Users\rs1\.vscode\extensions\ms-vscode-remote.remote-ssh-0.102.0\out\extension.js:1:578136)
	at t.handleInstallOutput (c:\Users\rs1\.vscode\extensions\ms-vscode-remote.remote-ssh-0.102.0\out\extension.js:1:581161)
	at t.tryInstall (c:\Users\rs1\.vscode\extensions\ms-vscode-remote.remote-ssh-0.102.0\out\extension.js:1:681881)
	at async c:\Users\rs1\.vscode\extensions\ms-vscode-remote.remote-ssh-0.102.0\out\extension.js:1:644110
	at async t.withShowDetailsEvent (c:\Users\rs1\.vscode\extensions\ms-vscode-remote.remote-ssh-0.102.0\out\extension.js:1:647428)
	at async t.resolve (c:\Users\rs1\.vscode\extensions\ms-vscode-remote.remote-ssh-0.102.0\out\extension.js:1:645160)
	at async c:\Users\rs1\.vscode\extensions\ms-vscode-remote.remote-ssh-0.102.0\out\extension.js:1:720916
[19:41:31.248] ------




[19:41:31.797] > ebebd4e708bc: start
> SSH_AUTH_SOCK====
> listeningOn====
> osReleaseId==windows==
> osVersion==10.0.22621==
> arch==x64==
> platform==windows==
> unpackResult====
> didLocalDownload==False==
> downloadTime====
> installTime====
> extInstallTime====
> serverStartTime==405==
> connectionToken==1aa1a1a1-1aa1-1111-aa11-1a1111a11111==
> ebebd4e708bc: end
> 16140, watching 3508
> gps : Cannot find a process with the process identifier 15780.
> At line:3 char:7
> + if (!(gps -Id $v_)) {
> +       ~~~~~~~~~~~
>     + CategoryInfo          : ObjectNotFound: (15780:Int32) [Get-Process], Proce
> ssCommandExceptio
>    n
>     + FullyQualifiedErrorId : NoProcessFoundForGivenId,Microsoft.PowerShell.Comm
> ands.GetProcessCo
>    mmand
> 
> server died, exit
[19:41:33.128] "install" terminal command done
[19:41:33.129] Install terminal quit with output: server died, exit

**Comment by tmsincomb** [comment]
Hi @bhosalems, I was able to fix this error `Found old VSCode package, cleaning up, failed to rm` by changing the remote servers default shell back to `bash` instead of `zsh`. I noticed my older centOS server was forcing vscode version 1.66.1 during remote setup which was a mismatch to my local side of 1.81.1. Since I know vscode fixed the zsh remote issue some time ago I can only assume remoting to a server with an older linux disto that is not compatible with a new vscode version, that isn't bash as default to run the installation of the remote vscode, is the issue.

I hope this helps.
